### DISCUSSION TITLE
Multithreading in v3.0

### DISCUSSION THREAD

**Discussion by Mwni** [explosion/spaCy#7203] [Help: Coding & Implementations]
This library seems multithreading compatible. When I share the same instance `nlp = load('...')` over multiple threads, there are no issues whatsoever. HOWEVER, it does not appear as if spawning more threads gives any performance benefits. I could imagine there being an internal lock that only allows for one thread at a time to work it, hence no multithreading gains.

I am seeking out clarification on this matter. Does each thread need its own instance for gains? Or is the only way via `n_process` on the pipe?

**Comment by svlandeg** [comment]
Hi! This is a good question for the discussions forum, so I'll move it there. You'll get a notification that this was closed, but the thread remains open on the forum.
